### PR TITLE
fix(TCOMP-2292): Studio side, TCK Record/Schema builder should support Object type value like JDBC ResultSet object

### PR DESCRIPTION
**Pull request by wwang-talend**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
https://jira.talendforge.org/browse/TCOMP-2292

### What does this PR adds (design/code thoughts)?
https://jira.talendforge.org/browse/TCOMP-2292

### CODE REVIEW

**Comment by wwang-talend** [comment]
TODO : add junit to DiRecordVisitorTest and DiRowStructVisitorTest

File: component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitor.java [hunk 1]
```diff
@@ -435,6 +435,8 @@ private void onArray(final Entry entry, final Collection value) {
     }
 
     private void onObject(final String name, final Object value) {
+        //in studio di, the object value is suitable to json string always? this code for construct the input tck record to tck processor, maybe is ok
+        //will adjust it if meet real requirement, then not toJson, only call "with(name, value)" api;
```
Comment (wwang-talend) [inline component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitor.java:439]: should call with(schema.getEntry("name"), value) api

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/RecordBuilderImplTest.java [hunk 2]
```diff
@@ -251,6 +251,23 @@ void decimal() {
...
+                .withEntry(new SchemaImpl.EntryImpl.BuilderImpl()
+                        .withName("object")
+                        .withNullable(true)
+                        .withType(Type.STRING)
+                        .withProp("talend.studio.type", "id_Object")
+                        .build())
+                .build();
+        final RecordImpl.BuilderImpl builder = new RecordImpl.BuilderImpl(schema);
+        Object value = new Object();
+        final Record record = builder.with(schema.getEntry("object"), value).build();
```
Comment (undx) [inline component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/RecordBuilderImplTest.java:267]: Try with some real world class...

File: component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitor.java (hunk 1 above)
Comment (undx) [reply component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitor.java:439]: jsonb may help us a lot for ser/deser. See my comment on slack.

**Comment by wwang-talend** [comment]
jsonb can't ser for jdbc resultset which have status and jsonb depend on get/is method, and some get/is method throw exception if status not match

File: component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitor.java [hunk 3]
```diff
@@ -435,6 +435,9 @@ private void onArray(final Entry entry, final Collection value) {
     }
 
     private void onObject(final String name, final Object value) {
+        // in studio di, the object value is suitable to json string always? this code for construct the input tck
+        // record to tck processor, maybe is ok
+        // will adjust it if not meet real requirement
         recordBuilder.withString(name, jsonb.toJson(value));
```
Comment (undx) [inline component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitor.java:441]: may not fit against what is done in the PR...

Comment (wwang-talend) [reply component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitor.java:441]: need to fix here for jdbc blob/clob type which map to id_Object, for example, suppose this all tck connector:
tmysqlinput(blob column)==>toracleoutput(blob column)
we can't call json.tojson here.

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**